File: backend/src/main/java/com/example/backend/persistence/entities/Rental.java
package com.example.backend.persistence.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Rental {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Car car;

    @ManyToOne
    private Customer customer;

    private LocalDate startDate;
    private LocalDate endDate;
    private float totalPrice;

    @Enumerated(EnumType.STRING)
    private RentalState state = RentalState.ACTIVE; // Default value set here

    public enum RentalState {
        CANCELED,
        ACTIVE,
        INVOICED
    }
}

File: frontend/src/interfaces/Rental.ts
/* eslint-disable @typescript-eslint/no-explicit-any */
import { Customer } from "./Customer";

export interface Rental {
    id?: number;
    customer: Customer;
    car: { id: number; model: string; pricePerDay: number }; // Car relationship
    startDate: string; // e.g., "2024-06-01"
    endDate: string;   // e.g., "2024-06-10"
    totalPrice: number;
    state?: any;
  }
  